include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -289,39 +298,40 @@ namespace alpaka
-                    ::alpaka::uniform_cuda_hip::detail::rtCheckLastError<true>(msg.c_str(), __FILE__, __LINE__);
+                    ::alpaka::uniform_cuda_hip::detail::rtCheckLastError<API, true>(msg.c_str(), __FILE__, __LINE__);
                 }
             }
         };
+
         //! The CUDA/HIP synchronous kernel enqueue trait specialization.
-        template<typename TAcc, typename TDim, typename TIdx, typename TKernelFnObj, typename... TArgs>
+        template<typename API, typename TAcc, typename TDim, typename TIdx, typename TKernelFnObj, typename... TArgs>
         struct Enqueue<
-            QueueUniformCudaHipRtBlocking,
-            TaskKernelGpuUniformCudaHipRt<TAcc, TDim, TIdx, TKernelFnObj, TArgs...>>
+            QueueUniformCudaHipRtBlocking<API>,
+            TaskKernelGpuUniformCudaHipRt<API, TAcc, TDim, TIdx, TKernelFnObj, TArgs...>>
         {
             ALPAKA_FN_HOST static auto enqueue(
-                QueueUniformCudaHipRtBlocking& queue,
-                TaskKernelGpuUniformCudaHipRt<TAcc, TDim, TIdx, TKernelFnObj, TArgs...> const& task) -> void
+                QueueUniformCudaHipRtBlocking<API>& queue,
+                TaskKernelGpuUniformCudaHipRt<API, TAcc, TDim, TIdx, TKernelFnObj, TArgs...> const& task) -> void
             {
                 ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
                 // TODO: Check that (sizeof(TKernelFnObj) * m_3uiBlockThreadExtent.prod()) < available memory idx
 
 #        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
                 // std::size_t printfFifoSize;
-                // cudaDeviceGetLimit(&printfFifoSize, cudaLimitPrintfFifoSize);
+                // API::deviceGetLimit(&printfFifoSize, API::limitPrintfFifoSize);
+                // std::cout << __func__ << "INFO: printfFifoSize: " << printfFifoSize << std::endl;
+                // API::deviceSetLimit(API::limitPrintfFifoSize, printfFifoSize*10);
+                // API::deviceGetLimit(&printfFifoSize, API::limitPrintfFifoSize);
                 // std::cout << __func__ << "INFO: printfFifoSize: " << printfFifoSize << std::endl;
-                // cudaDeviceSetLimit(cudaLimitPrintfFifoSize, printfFifoSize*10);
-                // cudaDeviceGetLimit(&printfFifoSize, cudaLimitPrintfFifoSize);
-                // std::cout << __func__ << "INFO: printfFifoSize: " <<  printfFifoSize << std::endl;
 #        endif
                 auto const gridBlockExtent = getWorkDiv<Grid, Blocks>(task);
                 auto const blockThreadExtent = getWorkDiv<Block, Threads>(task);
                 auto const threadElemExtent = getWorkDiv<Thread, Elems>(task);
 
-#        if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+#        if 1
                 dim3 const gridDim = uniform_cuda_hip::detail::convertVecToUniformCudaHipDim(gridBlockExtent);
                 dim3 const blockDim = uniform_cuda_hip::detail::convertVecToUniformCudaHipDim(blockThreadExtent);
 #        else
```

Comment:
Do we even need the #else branch anymore then?

Reply:
Nope.